base/stat.jl

```diff
@@ -26,6 +26,7 @@ export
     uperm
 
 struct StatStruct
+    desc    :: Union{AbstractString, OS_HANDLE}
```

Comment:
this will change `==` and `hash`, but I don't think people would have been relying on that?

Reply:
@staticfloat just a thought.. is the `StatStruct` hash ever used for artifact hashing etc.?

Reply:
GIven this broke existing symlink tests, I've defined custom hash and equality methods that skip `desc` and added a note to the struct